conan_provider.cmake

```diff
@@ -203,6 +203,11 @@ function(detect_host_profile output_file)
         string(APPEND PROFILE "tools.android:ndk_path=${CMAKE_ANDROID_NDK}\n")
     endif()
 
+    string(APPEND PROFILE "[buildenv]\n")
+    string(APPEND PROFILE "CC=${CMAKE_C_COMPILER}\n")
+    string(APPEND PROFILE "CXX=${CMAKE_CXX_COMPILER}\n")
+    string(APPEND PROFILE "LD=${CMAKE_LINKER}\n")
```

Comment:
Now Conan provides a ``compiler_executables`` configuration that might be a better mechanism to define this (it can work even with build systems that will not listen to CC/CXX env-vars). Setting the env-vars might not be necessarily bad, but we need to discuss the best approach.